course/deleteblock.php

```diff
@@ -60,16 +61,16 @@
 		//DB $query = "UPDATE imas_courses SET itemorder='$itemlist' WHERE id='{$_GET['cid']}'";
 		//DB mysql_query($query) or die("Query failed : " . mysql_error());
 		$stm = $DBH->prepare("UPDATE imas_courses SET itemorder=:itemorder WHERE id=:id");
-		$stm->execute(array(':itemorder'=>$itemlist, ':id'=>$_GET['cid']));
+		$stm->execute(array(':itemorder'=>$itemlist, ':id'=>$cid));
 
 		//DB mysql_query("COMMIT") or die("Query failed :$query " . mysql_error());
 		$DBH->commit();
 
 		$obarr = explode(',',$_COOKIE['openblocks-'.Sanitize::courseId($_GET['cid'])]);
 		$obloc = array_search($obid,$obarr);
 		array_splice($obarr,$obloc,1);
-		setcookie('openblocks-'.Sanitize::courseId($_GET['cid']),implode(',',$obarr));
-		header('Location: ' . $GLOBALS['basesiteurl'] . "/course/course.php?cid=".Sanitize::courseId($_GET['cid']));
+		setcookie('openblocks-'.Sanitize::courseId($_GET['cid']),implode(',',array_map('intval',$obarr)), 0, null, null, false, true);

```

Comment:
Use onlyInt.  Remove httponly flag - this cookie needs to be read and changed by Javascript